juju/juju
# Feature/lp 1847125

## Description of change

Add k8s level RBAC spec to allow charm to create 2nd set of RBAC resources;

## QA steps

- deploy charm has below podspec;

```yaml
serviceAccount:
  automountServiceAccountToken: true
  clusterRoleNames: ["coredns", "microk8s-hostpath"]
  rules:
    - apiGroups: [""]
      resources: ["pods"]
      verbs: ["get", "watch", "list"]
kubernetesResources:
  serviceAccounts:
    - name: rbac-foo
      automountServiceAccountToken: true
      clusterRoleNames: ["coredns", "microk8s-hostpath"]
      rules:
        - apiGroups: [""]
          resources: ["pods"]
          verbs: ["get", "watch", "list"]
```

- check rbac resources;

```bash
$ mkubectl get sa,roles,rolebindings -n<ns>
NAME                                  SECRETS   AGE
serviceaccount/mariadb-k8s            1         114s
serviceaccount/mariadb-k8s-operator   1         2m3s
serviceaccount/rbac-foo               1         114s

NAME                                                  AGE
role.rbac.authorization.k8s.io/mariadb-k8s            114s
role.rbac.authorization.k8s.io/mariadb-k8s-operator   2m3s
role.rbac.authorization.k8s.io/rbac-foo               114s

NAME                                                                  AGE
rolebinding.rbac.authorization.k8s.io/mariadb-k8s                     114s
rolebinding.rbac.authorization.k8s.io/mariadb-k8s-coredns             114s
rolebinding.rbac.authorization.k8s.io/mariadb-k8s-microk8s-hostpath   114s
rolebinding.rbac.authorization.k8s.io/mariadb-k8s-operator            2m3s
rolebinding.rbac.authorization.k8s.io/rbac-foo                        113s
rolebinding.rbac.authorization.k8s.io/rbac-foo-coredns                113s
rolebinding.rbac.authorization.k8s.io/rbac-foo-microk8s-hostpath      112s
``` 


## Documentation changes

Yes

## Bug reference

https://bugs.launchpad.net/juju/+bug/1847125
